### ISSUE TITLE
App works fine in Ripple Emulator but i get White Screen of Death in Android Emulator

### ISSUE THREAD

**Issue by ice423cube** [signalpoint/DrupalGap#200]
I am able to run my app fine in the ripple emulator in chrome.  When I try to switch to eclipse and run it through eclipse with the emulator I just get a white screen of death.  

Any tips on switching code back to emulator that I might not be aware of?

Any way to see if drupalgap is even loading? 

I have tried so many things at this point I have lost track and hope.  

Thanks for your help.

**Comment by signalpoint** [comment]
If you haven't reviewed them, there are troubleshooting topics here:

http://drupalgap.org/troubleshoot
https://drupal.org/node/2015065

Next, I'd recommend just following the Eclipse Hello World for PhoneGap 2.9.0 and make sure that can run by itself in the Android emulator. Once that is working, bring DrupalGap into the PhoneGap project.

Also, make sure your AndroidManifest.xml file is using, for example:

android:minSdkVersion="7" android:targetSdkVersion="16"

Either way, for sure the best approach is just getting the PhoneGap 2.9.0 Hello World working in Eclipse.

Additionally, make sure LogCat is enabled in Eclipse and you should be able to see debug messages output to the console. You can set drupalgap.settings.debug = true to see debug info.

**Comment by ice423cube** [comment]
I just wanted to update for anyone else that might be having this issue.  It turns out that when I was using the emulator it didn't' have internet there for it would fail with this error, "Application Error: The connection to the server was unsuccessful... index.html".  I opened up a browser in the emulator and no pages would load.  I then followed this tip on stackoverflow.com (http://stackoverflow.com/questions/2437366/android-emulator-internet-access).  This worked and my app loaded.  Hope that helps anyone else trying to get an app working using phonegap, drupalgap and drupal.

**Comment by deltaag** [comment]
As I am using quickstart for my Drupal development, and I recently upgraded to Ubuntu 14.04, to get drupalgap working in Android emulator, I needed to complete a few steps. Hope this helps someone!

First we need to define a port. Add this line to `/home/quickstart/websites/config/ports.conf` (this will add it to final destination on `/etc/apache2/ports.conf`):

```
listen 55555
```

Then we need to add the port to apache’s virtual host. Edit `/etc/apache2/sites-available/YourSite.dev.conf` file and copy all `<VirtualHost *:80> … </VirtualHost>` section and paste it in the button of the file and change the first line to:

```
<VirtualHost *:55555>
```

Then we need to restart apache by executing this command in terminal:

```
sudo services apache2 restart
```

Now our dev website will be available on http://localhost:55555
For Android emulator we will need to use http://10.0.2.2:55555 to access our website. So in drupalgap application, in `MyApp/www/app/settings.js` set `Drupal.settings.site_path` to http://10.0.2.2:55555 and the dev website will be available for drupalgap.
Enjoy!
